kategory/src/main/kotlin/kategory/arrow/Function1.kt

```diff
@@ -6,20 +6,19 @@ fun <P, R> ((P) -> R).k(): Function1<P, R> =
         Function1(this)
 
 @Suppress("UNCHECKED_CAST")
-fun <R> Function1F<R>.ev(): FunctionInject<R> =
-        this as FunctionInject<R>
+fun <P, R> Function1F<R>.ev(): (P) -> R =
+        (this as Function1<P, R>).f
```

Comment:
Look at this function. The P is only part of the output, so it has to be specified at callsite as the compiler inference doesn't necessarily pick it up.